code/modules/mob/living/holders.dm

```diff
@@ -46,6 +46,12 @@
 	else if(istype(loc,/turf) || !(contents.len))
 		return qdel(src)
 
+/obj/item/weapon/holder/dropped()
+	..()
+	spawn(1) // im a cheater
+		if(istype(loc,/turf) || !(contents.len))
+			qdel(src)
```

Comment:
Prepare for unforeseen consequences.

Reply:
devilspawn

Reply:
I coded this months ago and don't remember why I did that but I'm sure it had a purpose

Reply:
This actually does make sense. What this is doing is destroying the holder itself if it no longer contains a mob (obvious purpose) or if it is placed on the ground (this causes the holder to delete, dropping the mob on the ground). The spawn is a little odd though. What purpose?

Reply:
I added this so that the holder would immediately delete itself when dropped, instead of waiting for process(). I coded this months ago so I forgot exactly why I added the spawn. I think it was because there was something occasionally causing the mob to end up in nullspace and adding spawn was a lazy fix, but I'll have to test to make sure.